src/UserFilter.php

```diff
@@ -23,22 +23,37 @@
  */
 class UserFilter implements FilterInterface
 {
+    /**
+     * Caseless matching of masks. True for caseless, false for case to matter.
+     *
+     * @var boolean
+     */
+    protected $caseless = false;
+
     /**
      * List of masks identifying users from whom to forward events
      *
      * @var array
      */
-    protected $masks;
+    protected $masks = [];
 
     /**
-     * Accepts a list of masks identifying users from whom to forward events.
+     * Exception code used when caseless is not set as a boolean
+     */
+    const ERR_CASELESS_BOOLEAN = 1;
+
+    /**
+     * Accepts filter configuration.
      *
-     * @param array $masks
+     * @param array $masks An array of masks identifying users from whom to forward events
+     * @param boolean $caseless True to use the caseless preg modifier when comparing masks,
+     *        false by default
      * @see http://www.ircbeginner.com/opvinfo/masks.html
      */
-    public function __construct(array $masks)
+    public function __construct(array $masks, $caseless = false)
     {
         $this->masks = $masks;
+        $this->caseless = $caseless;
```

Comment:
I'd typecast `$caseless` to `bool` here. Since the plugin presently supports PHP 5.6, that's the closest we can get to using a `bool` type hint.